packages/features/form-builder/FormBuilder.tsx

```diff
@@ -498,6 +493,56 @@ function FieldEditDialog({
                         );
                       }}
                     />
+                    <div className="mt-6">
+                      <div className="flex items-center gap-1">
+                        <label className="text-emphasis block text-sm font-medium">Identifier</label>
+                        <Link
+                          href="https://cal.com/docs/core-features/event-types/booking-questions#booking-field-identifier"
+                          target="_blank">
+                          <Tooltip content="Click to learn more ↗" side="top">
+                            <Info className="hover:bg-emphasis h-5 w-5 cursor-pointer rounded p-0.5" />
+                          </Tooltip>
+                        </Link>
+                        {!identifierFieldVisible && identifierBadge !== "" && (
+                          <Badge variant="grayWithoutHover" className="text-subtle">
+                            {identifierBadge?.replaceAll(".", "-").replaceAll(" ", "-")}
+                            <Tooltip content="Edit identifier" side="top">
+                              <div
+                                data-testid="edit-identifier"
+                                onClick={() => {
+                                  fieldForm.setValue(
+                                    "name",
+                                    identifierBadge?.replaceAll(".", "-").replaceAll(" ", "-")
+                                  );
+                                  setIdentifierFieldVisible(true);
+                                }}>
+                                <Edit className="hover:text-emphasis h-4 w-4 cursor-pointer" />
+                              </div>
+                            </Tooltip>
+                          </Badge>
+                        )}
+                      </div>
+                      {identifierFieldVisible && (
+                        <>
+                          <InputField
+                            required
+                            {...fieldForm.register("name")}
+                            onChange={(e) => {
+                              fieldForm.setValue("name", getFieldIdentifier(e.target.value || ""));
+                            }}
+                            disabled={
+                              fieldForm.getValues("editable") === "system" ||
+                              fieldForm.getValues("editable") === "system-but-optional"
+                            }
+                            label={t("identifier")}
+                            labelClassName="invisible -mb-2"
+                          />
+                          <p className="text-subtle text-sm">
+                            You will not be able to edit these later. Letters, numbers & hyphens only.
```

Comment:
This is not true, identifier can always be edited. But yeah, it is recommended to not edit it because if you do then any webhook or prefill param using this identifier has to be changed.


Maybe we can change the text to 'It is not recommended to change these' ?

Reply:
Plus the text needs to use translation key